src/spot_skills/src/spot_skills/follow_trajectory_server.py

```diff
@@ -0,0 +1,93 @@
+"""Define a control_msgs/FollowJointTrajectory action server for Spot's arm.
+
+Reference:
+https://docs.ros.org/en/noetic/api/control_msgs/html/action/FollowJointTrajectory.html
+"""
+
+from actionlib import SimpleActionServer
+from control_msgs.msg import (
+    FollowJointTrajectoryAction,
+    FollowJointTrajectoryActionGoal,
+    FollowJointTrajectoryActionResult,
+)
+from rospy import loginfo
+
+from spot_skills.joint_trajectory import JointTrajectory
+from spot_skills.spot_arm_controller import ArmCommandOutcome, SpotArmController
+
+
+class SpotFollowJointTrajectoryServer:
+    """An action server that controls Spot's arm to follow a trajectory."""
+
+    def __init__(self, action_name: str, arm_controller: SpotArmController) -> None:
+        """Initialize the ROS node for the FollowJointTrajectory action server.
+
+        :param    action_name       Name used to access the action server through ROS
+        :param    arm_controller    Interface used to control Spot's arm
+        """
+        self._action_name = action_name
+        self._arm_controller = arm_controller
+        self._result = None
+
+        self._action_server = SimpleActionServer(
+            self._action_name,
+            FollowJointTrajectoryAction,
+            execute_cb=self.callback,
+            auto_start=False,
+        )
+        self._action_server.start()
+
+        loginfo(f"{self._action_name}: Action server has started.")
+
+    def execute_callback(self, goal: FollowJointTrajectoryActionGoal) -> None:
+        """Execute the action by controlling Spot's arm to follow the given trajectory.
+
+        :param      goal        Joint trajectory to be followed
+        """
+        # Extract all fields of the received action goal message
+        trajectory = JointTrajectory.from_ros_msg(goal.trajectory)
+
+        # TODO: The ArmController could use the below joint tolerances to enforce safe
+        #   execution of the trajectory, within some position/velocity tolerance.
+        # The same logic would allow the server to publish feedback to its client(s)
```

Comment:
Could add joint tolerances and feedback to the action, but I suspect that MoveIt doesn't require it.

Reply:
Unless it becomes clear that MoveIt requires this feedback, see #15 for this effort.